### ISSUE TITLE
A simple idea to solve the  error : "rpn_loss_box: nan"

### ISSUE THREAD

**Issue by yaoing** [endernewton/tf-faster-rcnn#284]
There is an error with bounding boxes' coordinates,when some of coordinate values  become 65535.

So, we can open the file named "minibatch.py" , find the line:
`
gt_boxes = np.empty((len(gt_inds), 5), dtype=np.float32)

`
and insert some codes in front of it :
`
boxes = roidb[0]['boxes'][gt_inds, :]
for idx in range(len(boxes)):
    boxes[idx][boxes[idx]==65535] = 0
`

这可能算不上是个好的解决方案哈 ~

**Comment by endernewton** [comment]
the underlying reason might be your dataset is 0-based rather than 1-based in coordinates?

**Comment by chichivica** [comment]
I faced that issue when forget about **-1** for x_max and y_max in dataset loader
![](https://i.imgur.com/GPvYjKr.png)

**Comment by FMsunyh** [comment]
I also get this issue. And I have deleted the -1. But it still happen!
Does it have a better solution now?

![image](https://user-images.githubusercontent.com/9877644/41644942-26dc5d3a-74a2-11e8-858e-3b25719ed70e.png)

**Comment by kinzzoku** [comment]
@FMsunyh Did you delete the .pkl files created in in the cache folder of the dataset?
For me deleting the -1 didn't have any effect too at first. But deleting the -1 and after that the .pkl files solved it for me.

**Comment by FMsunyh** [comment]
@kinzzoku yes, I've deleted .pkl file and rebuild it again, but same issue with me.

**Comment by dby2017** [comment]
> There is an error with bounding boxes' coordinates,when some of coordinate values become 65535.
> 
> So, we can open the file named "minibatch.py" , find the line:
> 
> ```python
> gt_boxes = np.empty((len(gt_inds), 5), dtype=np.float32)
> ```
> 
> and insert some codes in front of it :
> 
> ```python
> boxes = roidb[0]['boxes'][gt_inds, :]         
> for idx in range(len(boxes)):
>   &nbsp; boxes[idx][boxes[idx]==65535] = 0
> ```
> 
> 这可能算不上是个好的解决方案 ~
  &nbsp; boxes[idx][boxes[idx]==65535] = 0 请问这样做是为什么？

**Comment by NaJongHo** [comment]
Hello.I have the same problem.

I changed the code as described in the following figure.

![nbsp](https://user-images.githubusercontent.com/17251247/53783831-9a4d3800-3f56-11e9-9105-e74692251711.png)

The first error was caused by &npsp.
The error picture is as follows:
![nbsp_error](https://user-images.githubusercontent.com/17251247/53783857-b2bd5280-3f56-11e9-81f9-43bba93b756e.png)


So I erased $npsp and compiled the code.
![code](https://user-images.githubusercontent.com/17251247/53783878-c8cb1300-3f56-11e9-8e8e-06a34ab8a371.png)

However, a nan value is still generated.
![some](https://user-images.githubusercontent.com/17251247/53783942-f1eba380-3f56-11e9-9fd0-f3e1186be6a0.png)

Please help me with the solution.

**Comment by ZhuanShan** [comment]
> Hello.I have the same problem.
> I changed the code as described in the following figure.
> 
> The first error was caused by &nbsp.
> The error picture is as follows:
> 
> So I erased $nbsp and compiled the code.
> 
> However, a nan value is still generated.
> 
> Please help me with the solution.

hi, have you found the soluntion? I meet the same problem now. could you please help me?

**Comment by ZhuanShan** [comment]
@NaJongHo hi, I got the same problem. Did you fix it? could you please help me?
